target/scripts/startup/setup.d/getmail.sh

```diff
@@ -4,38 +4,59 @@ function _setup_getmail() {
   if [[ ${ENABLE_GETMAIL} -eq 1 ]]; then
     _log 'trace' 'Preparing Getmail configuration'
 
-    local GETMAILRC ID CONFIGS
+    # Verify correct value for GETMAIL_POLL. Valid are any numbers greater than 0.
+    if ! [[ ${GETMAIL_POLL} =~ ^[0-9]+$ && ${GETMAIL_POLL} -gt 0 ]]; then
+      _log 'warn' "Invalid value for GETMAIL_POLL: ${GETMAIL_POLL}"
+      _log 'warn' "Getmail will be disabled"
+      # return 1
+    fi
```

Comment:
> What's going on with the return 1 comment here?

I just forgot to remove that comment.

There is no easy way to disable (or preventing to start) the getmail service at this point. So I decided to let the setup function complete (there may also be more than one warning) and once the service starts and notices the same, it will disable itself.

The benefit of that is, that the output (the warning) from the setup function goes to STDOUT and is visible for the user via `docker logs`. In comparison, the service output is not redirected to STDOUT, thus not directly visible. One have to check `/var/log/supervisor/getmail.log`.

Reply:
> In comparison, the service output is not redirected to STDOUT, thus not directly visible.

Ah right.

You could get this into the actual mail log if you do not use our log helper (_or the log helper were to use `logger` instead_): https://github.com/docker-mailserver/docker-mailserver/issues/3561#issuecomment-1916165908

> ```console
> $ logger --priority mail.error --id=1337 --tag dms-changedetector "This is an example error message!"
> 
> # It would be processed via rsyslog to a log line like this (but with the timestamp prepended):
> mail dms-changedetector[1337]: This is an example error message!
> ```

Might be more appropriate for a service wrapper to use that for logs since other services use syslog too and generally have little stdout output for supervisor to capture 🤔

Reply:
> You could get this into the actual mail log

At this point, those error messages are already in the mail.log from the setup function. So `_log` to the supervisor log files is fine.
And if getmail encounters an error, it will send the message to syslog => `mail.log` due to `message_log_syslog = true` set.

Reply:
I like the `logger` idea.. I'll remove the error checking from the setup function and use the logger command in the service file 👍

Reply:
> > You could get this into the actual mail log
> 
> At this point, those error messages are already in the mail.log from the setup function. So `_log` to the supervisor log files is fine.

I don't recall if we had `_log` calls go into `mail.log`. We output to stdout/stderr for docker logs:

Reply:
> I don't recall if we had _log calls go into mail.log.

My mistake. What I mean was, it will end in STDOUT and is available via `docker logs`. Because we also tail `mail.log` to STDOUT, I mixed things up.